.github/actions/nss/action.yml

```
@@ -1,19 +1,53 @@
 name: Fetch and build NSS
 
+inputs:
+  type:
+    description: 'Whether to do a debug or release build of NSS'
+    required: true
+    default: 'Debug'
+
+# This step might be removed if the distro included a recent enough
+# version of NSS.  Ubuntu 20.04 only has 3.49, which is far too old.
+# (neqo-crypto/build.rs would also need to query pkg-config to get the
+# right build flags rather than building NSS.)
+#
+# Also see https://github.com/mozilla/neqo/issues/1711
+
 runs:
   using: composite
   steps:
-    - name: Fetch and build NSS
+    - name: Checkout NSPR
+      uses: actions/checkout@v4
+      with:
+        repository: 'nss-dev/nspr'
+        path: ${{ github.workspace }}/nspr
+
+    - name: Checkout NSS
+      uses: actions/checkout@v4
+      with:
+        repository: 'nss-dev/nss'
+        path: ${{ github.workspace }}/nss
+
+    - name: Build
       shell: bash
       run: |
-        git clone --depth=1 https://github.com/nss-dev/nspr "$NSPR_DIR"
-        git clone --depth=1 https://github.com/nss-dev/nss "$NSS_DIR"
-        $NSS_DIR/build.sh -g -Ddisable_tests=1 -o
+        # We want to do an optimized build for accurate CPU profiling, but
+        # we also want debug symbols for that, which the normal optimized NSS
+        # build process doesn't provide.
+        [ "${{ runner.os }}" != "Windows" ] && export CFLAGS=-ggdb3
+        if [ "${{ inputs.type }}" != "Debug" ]; then
+          export OPT="-o"
+          export NSS_TARGET=Release
+        else
+          export NSS_TARGET=Debug
+        fi
+        $NSS_DIR/build.sh -g -Ddisable_tests=1 $OPT --static
+        echo "NSS_TARGET=$NSS_TARGET" >> "$GITHUB_ENV"
+        export NSS_OUT="$NSS_DIR/../dist/$NSS_TARGET"
```

```suggestion
        NSS_OUT="$NSS_DIR/../dist/$NSS_TARGET"
```

I'm fairly sure that you don't need `export` if you are using the string in this shell.